Sources/BraveWallet/Settings/DappsSettings.swift

```diff
@@ -3,58 +3,144 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import Foundation
-import BraveUI
 import SwiftUI
-import struct Shared.Strings
-
-struct ManageSiteConnectionsView: View {
+import BraveShared
+import BraveCore
 
+struct DappsSettings: View {
+  var coin: BraveWallet.CoinType
   @ObservedObject var siteConnectionStore: ManageSiteConnectionsStore
+  @ObservedObject var defaultWallet: Preferences.Option<Int>
+  @ObservedObject var allowProviderAccess: Preferences.Option<Bool>
   @State private var filterText: String = ""
   @State private var isShowingConfirmAlert: Bool = false
   
+  init(
+    coin: BraveWallet.CoinType,
+    siteConnectionStore: ManageSiteConnectionsStore
+  ) {
+    self.coin = coin
+    self.siteConnectionStore = siteConnectionStore
+    switch coin {
+    case .eth:
+      self.defaultWallet = Preferences.Wallet.defaultEthWallet
+      self.allowProviderAccess = Preferences.Wallet.allowEthProviderAccess
+    case .sol:
+      self.defaultWallet = Preferences.Wallet.defaultSolWallet
+      self.allowProviderAccess = Preferences.Wallet.allowSolProviderAccess
+    default:
+      self.defaultWallet = Preferences.Option(key: "Unsupported-Coin-Default-Wallet", default: -1)
+      self.allowProviderAccess = Preferences.Option(key: "Unsupported-Coin-Allow-Access", default: false)
```

Comment:
creating those fake prefs seems weird, any way to avoid it?

Reply:
fixed in [d43c8d5](https://github.com/brave/brave-ios/pull/5991/commits/d43c8d5e53c5dc80c56a7b986e4d6e208c049105)